cli/azd/docs/external-prompting.md

```diff
@@ -0,0 +1,97 @@
+
+The body of the request contains a JSON object with all the information about the prompt that `azd` needs a response for:
+
+```jsonc
+{
+    "type": "<string>", // one of "string", "password", "select", "multiSelect", "confirm", "directory"
+    "options": {
+      "message": "<string>", // the message to be displayed as part of the prompt
+      "help": "<string>", // optional help text that can be displayed upon request
+      "choices": [  { "value": "string", "detail": "string" | undefined } /* ... */  ], // for select and multiSelect types - the valid options the user should pick from
+      "defaultValue":  "<string>" | "<string>[]" | boolean, // optional default value (or values, for multiSelect), when undefined there is no default.
+    }
+}
+```
+
+The `password` type represents a string value which represents a password. The host may want to use a different UI element (perhaps one that uses `***` instead of characters) when prompting.
+
+The server should respond with 200 OK and the body should be one of the of the following three shapes:
+
+### Success 
+
+When the host is able to prompt the value it is returned with a `status` of `success`:
+
+```jsonc
+{
+    "status": "success", 
+    "value": "<string>" | "<string>[]"
+}
+```
+
+When the type is `confirm` the value should be either `"true"` or `"false"` (a string, not a JSON boolean) indicating if the user confirmed the operation (`"true"`) or rejected it (`"false"`). Note that a user rejecting a confirm prompt still results in a `"success"` status (the value is simply `"false"`). In the case of `multiSelect` an array of string values is returned, each individual value is a value from the `"options"` array that was selected by the user.
+
+### Cancelled
+
+The user may decline to provide a response (imagine hitting a cancel button on the dialog that is being used to present the question, or a user hitting something like CTRL+C in a terminal interaction to abort the question asking but not the entire application).
+
+In this case the `status` is `cancelled`.
+
+```jsonc
+{
+    "status": "cancelled", 
+}
+```
+
+`azd` returns a special `Error` type internally in this case, which up-stack code can use.
+
+### Error
+
+Some error happened during prompting - the error message will be returned as is.
+
+```jsonc
+{   "status": "error",
+    "message": "<string>" // an error message, will be the text of the `Error` returned inside `azd`.
+}
+```
+
+Note that an error prompting leads to a successful result at the HTTP layer (200 OK) but with a special error object. `azd` treats other responses as if the server has an internal bug.
+
+## Open Issues
+
+- [ ] Some hosts, such as VS, may want to collect a set of prompts up front and present them all on a single page as part of an end to end - how would we support this? It may be that the answer is "that's a separate API" and this solution is simply focused on "when `azd` it self is driving and end to end workflow".
```

Comment:
+1 on separate API for now. From pragmatic standpoint, it'd be nice if the API just reused the same contracts we set here except in a "batching manner.